File: example/sendProjectData.js
var besc_client = require("../src/index");

var dateConvert = require("../src/utils/dateConvert");

var keypair = new besc_client.keyPair("1", "abc123");

const ProjectData = besc_client.ProjectData;
const Device = besc_client.Device;

//var host_client = new besc_client.Host("http://localhost:3001");
var host_client = besc_client.Host.createDefault();

(async () =>{ 

    var projectData = new ProjectData(
        dateConvert.formatDatetimeString("2019-05-29T06:00:00"),
        "Testing", 
        [
            new Device("AABC1",40), 
            new Device("AABC2",70)
        ],
        110,
        80.5,
        "101.1212, 112.1133"
    );

    /*
    var projectData = ProjectData.creatWithCurrentTime(
        "Testing", 
        [
            new Device("AABC1",40), 
            new Device("AABC2",70)
        ],
        110,
        80.5,
        "101.1212, 112.1133"
    );
    */

    try{
        var response = await besc_client.API.sendProjectData(host_client, keypair, projectData);

        //console.log(response);
    }
    catch(apiError){
        // apiError.statusCode - request http status code
        // apiError.error - error message that send back from server
        //console.log(apiError);
    }

})();

File: example/sendProjectData.py
from src.utils import dateConvert
from src.api import API
from src.model.project_data import ProjectData
from src.model.device import Device
from src.services import besc_host as Host
from src.model import keypair as keyPair
import datetime

keypair = keyPair.KeyPair("1", "gS5pBilMCBS91z0go3LO6iqLUfZCU0DH")

host_client = Host.BESC_ESS_Host("http://carboapi.besc.online/besc-data")

def sendProjectData():

    projectData = ProjectData(
        "Testing",
        datetime.datetime.now(),
        [
            Device("AABC1", 40),
            Device("AABC2", 70)
        ],
        110,
        80,
        "101.1212, 112.1133"
        )

    try:
        response = API.sendProjectData(keypair, projectData)

    except Exception as e:
        raise (response)
        print(e)

if __name__ == "__main__":
    sendProjectData()
